capture-sdk/sdk/src/main/java/net/gini/android/capture/internal/camera/photo/PhotoCropModifier.java

```diff
@@ -0,0 +1,91 @@
+package net.gini.android.capture.internal.camera.photo;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Matrix;
+import android.graphics.Rect;
+import android.util.DisplayMetrics;
+
+import java.io.ByteArrayOutputStream;
+
+public class PhotoCropModifier implements PhotoModifier {
+
+    private final Photo mPhoto;
+    private final int[] mScreenSize;
+    private final Rect aRect;
+
+    public PhotoCropModifier(Photo mPhoto, int[] mScreenSize, Rect aRect) {
+        this.mPhoto = mPhoto;
+        this.mScreenSize = mScreenSize;
+        this.aRect = aRect;
+    }
+
+    @Override
+    public void modify() {
+
+        if (mPhoto.getData() == null)
+            return;
+
+        synchronized (aRect) {
+            byte[] originalBytes = mPhoto.getData();
+            try {
+                float rotation = 0;
+
+                switch (mPhoto.getRotationForDisplay() % 360) {
+                    case 90:
+                        rotation = 90;
+                        break;
+                    case 180:
+                        rotation = 180;
+                        break;
+                    case 270:
+                        rotation = 270;
+                        break;
+                    default:
+                        break;
+                }
+
+                Matrix matrix = new Matrix();
+                matrix.postRotate(rotation);
+
+                Bitmap originalBitmap = BitmapFactory.decodeByteArray(mPhoto.getData(),
+                        0, mPhoto.getData().length);
+
+                Bitmap rotatedBitmap = Bitmap.createBitmap(originalBitmap, 0, 0, originalBitmap.getWidth(),
+                        originalBitmap.getHeight(), matrix, false);
+
+
+                int x1 = rotatedBitmap.getWidth() * aRect.left / mScreenSize[0];
+                int y1 = rotatedBitmap.getHeight() * aRect.top / mScreenSize[1];
+                int width1 = rotatedBitmap.getWidth() * aRect.width() / mScreenSize[0];
+                int height1 = rotatedBitmap.getHeight() * aRect.height() / mScreenSize[1];
+
+                matrix.postRotate(finalRotation(rotation));
```

Comment:
[Suggestion] If I'm not mistaken then this line reverses the rotation so that the cropped image has the same rotation as the original.

I did a bit of research and found that we can use `matrix.invert(matrix)` to reverse the rotation.

With `finalRotation()` the orientation was wrong when I tested a 270 rotation on a Galaxy S21 by changing the line 45 in `ActivitHelper` to `activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT)`.

With `matrix.inverse(matrix)` it worked for 270 and other rotations as well (I uncommented the lines in `ActivityHelper.lockToPortraitOrientation()` to be able to test landscape orientations on the phone).

Reply:
Awesome thanks.